website/server/models/group.js

```diff
@@ -677,6 +698,11 @@ schema.methods.startQuest = async function startQuest (user) {
 
   // Changes quest.members to only include participating members
   this.quest.members = _.pickBy(this.quest.members, _.identity);
+
+  // Persist quest.members early to avoid simultaneous handling of accept/reject
+  // while processing the rest of this script
+  this.update({ $set: { 'quest.members': this.quest.members } }).exec();
```

Comment:
I'm a bit confused about this update here, I mean which bug would it prevent

Reply:
The issue states that another cause of bugs is simultaneous force-start call with an accept/reject call. Previously the `quest.members` list is updated here to only have members who accepted the invite, then it goes on to do a lot of work updating each of the users etc. so it could take a while, until finally `group.save()` is called in the controller which is when the members list gets persisted to db. In the mean time if someone else makes accept/reject call it would update the db, but then it would get overridden by the `group.save()` of the force-start call.

And there is also the potential of executing all this startQuest logic twice simultaneously (although it's pretty unlikely), if the force-start is called at the same time as the last member to accept/reject (which then goes on to execute startQuest also). This change should make that even less likely by updating the db early before all the user updates. Because then if another accept/reject call comes in while doing the user updates, the member would already be removed from the `quest.members` list and so the query will not match and they get an error.